### PR TITLE
Fix new auto enabled ruff rules

### PR DESCRIPTION
**Pull request by jelly**
As this effects `testlib` let's run one full test run, but not all distro's that seems a bit excessive.

### CODE REVIEW

File: test/verify/ruff.toml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 extend = "../common/ruff.toml"
 ignore = [
     "B023",     # Function definition does not bind loop variable
+    "PT027",    # Use `pytest.raises` instead of unittest-style `assertRaises`
```
Comment (allisonkarlitskaya) [inline test/verify/ruff.toml:4]: Thanks for not adding this at the toplevel.

**Review by allisonkarlitskaya — APPROVED** [review]
Looks good, thanks.  The unit test fail was a network flake, so retrying that.

**Review by allisonkarlitskaya — CHANGES_REQUESTED** [review]
```
#   Cause: Failed to parse `/source/test/verify/ruff.toml`: TOML parse error at line 4, column 5
```

that's not a flake, though...

**Comment by jelly** [comment]
So I did not consume enough :coffee: today, now tox passes but the unit test container was not updated yet to have the new ruff.

File: test/verify/ruff.toml [hunk 2]
```diff
@@ -1,4 +1,5 @@
 extend = "../common/ruff.toml"
 ignore = [
     "B023",     # Function definition does not bind loop variable
+    "PT",       # HACK: ignore all of PT as our tasks container ruff does not know about PT027 yet
```
Comment (mvollmer) [inline test/verify/ruff.toml:4]: I don't like this.  What happens when we ignore just "PT027"?

Comment (mvollmer) [reply test/verify/ruff.toml:4]: Ah, the mommit message has a better motivation: "Our integration tests are not using pytest but unittest.TestCase so we want to silence these rules." That makes sense to me, but then the comment shouldn't say that we only want to ignore "PT027".

Comment (jelly) [reply test/verify/ruff.toml]: Yes, adjusted the comment.

**Review by allisonkarlitskaya — APPROVED** [review]
This was my preferred solution :)

**Pull request merged**